locals.tf

```
@@ -1,4 +1,4 @@
 locals {
-  env      = lower(terraform.workspace)
+  env      = var.environment != "" ? var.environment : lower(terraform.workspace)
```

I think this conditional is more understandable as:
`env = var.environment == "" ? lower(terraform.workspace) : var.environment`

---

Don't have a preference so I'm happy to switch it

---

You can use a [coalesce](https://developer.hashicorp.com/terraform/language/functions/coalesce) statement instead

```suggestion
  env      = coalesce(var.environment, lower(terraform.workspace))
```

Here is an example using `terraform console`

```bash
> coalesce("", lower(terraform.workspace))
"default"
> coalesce("myenv", lower(terraform.workspace))
"myenv"
```

---

nice - I didn't know about the coalesce function

---

Oh cool, I've not seen that one before